packages/n/nelib/xmake.lua

```diff
@@ -0,0 +1,59 @@
+package("nelib")
+    set_kind("library")
+    set_homepage("https://github.com/hexne/nelib")
+    set_description("A commonly used tool collection library")
+
+    add_urls("https://github.com/hexne/NeLib.git")
+    add_versions("2024.11.15", "ee9195d2b2612da06e0ac4aaa1f6c7cb6c94699a")
+
+    add_configs("modules", {description = "Build with C++20 modules support.", default = false, type = "boolean"})
+    add_configs("header_only", {description = "Build as a headeronly library.", default = false, type = "boolean"})
+    set_policy("package.cmake_generator.ninja", true)
+
+    add_deps("opencv")
+
+
+    on_load(function (package)
+        if not package:config("modules") then
+            package:add("deps", "cmake")
+            if package:config("header_only") then
+                package:set("kind", "library", {headeronly = true})
+            end
+        end
+    end)
+
+    on_install("linux", function (package)
+        if not package:config("modules") then
+            os.cp("*.h", package:installdir("include"))
+            if not package:config("header_only") then
+                io.replace("CMakeLists.txt", "NOT GLM_DISABLE_AUTO_DETECTION", "FALSE")
+                local configs = {"-DGLM_BUILD_TESTS=OFF"}
+                table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+                import("package.tools.cmake").build(package, configs)
+                os.cp("**.pcm", package:installdir("include"))
+                os.cp("**.a", package:installdir("lib"))
+            end
+        else
+            io.writefile("xmake.lua", [[ 
+                add_requires("opencv")
+                target("nelib")
+                    add_packages("opencv")
+                    set_kind("$(kind)")
+                    set_languages("c++20")
+                    add_headerfiles("./**.h", {outputdir = 'nelib'})
+                    add_includedirs(".")
+                    add_files("**.cppm", {public = true})
+            ]])
+            import("package.tools.xmake").install(package)
+        end
+
+    end)
+
+
+    on_test(function (package)
+        assert(package:check_cxxsnippets({test = [[
+            void test() {
+                ;
```

Comment:
no any test code

Reply:
in on_test, can't test module,even if the library can be installed correctly
such as :
```
    on_test(function (package)
        assert(package:check_cxxsnippets({test = [[
            import Image;
            void test() {
                nl::Image image("");
            }
        ]]}, {configs = {languages = "c++20", includes = "nelib/Image.cppm"}}))
    end)
```
but i will get error : 
```
/tmp/.xmake1000/241115/_23D83C3EEA4B450697124F07AE0D9444.cpp:4:20: fatal error: module 'Image' not found
    4 |             import Image;
      |             ~~~~~~~^~~~~
1 error generated.
```
If I don't test the content in on_test, then the library can actually be used.
src/mian.cpp
```c++
import Image;

int main(int argc, char** argv) {

    nl::Image image("../../../../lqb.png");
    image.show_and_wait("show");
    return 0;
}
```
use ``xmake f --toolchain=clang -c && xmake && xmake run``, This will give you the right result
```
☁  test xmake f --toolchain=clang -c && xmake 
checking for platform ... linux
checking for architecture ... x86_64
[  0%]: <test> generating.module.deps src/main.cpp
[ 90%]: compiling.release src/main.cpp
[ 95%]: linking.release test
[100%]: build ok, spent 1.699s
warning: std and std.compat modules not found! maybe try to add --sdk=<PATH/TO/LLVM> or install libc++
```